### ISSUE TITLE
bweuler - Euler number for binary images

### ISSUE THREAD

**Issue by juliohm** [JuliaImages/Images.jl#629]
Do you recommend a paper discussing an efficient implementation? I see that `bwlabel` is already implemented in `Images.jl` as `label_components`. I would like to contribute `bweuler` if I can find the time. Can the Euler number be implemented in terms of the `label_components` output?

**Comment by timholy** [comment]
Yes, I would call the function `euler_characteristic` (there are [too many](https://en.wikipedia.org/wiki/Euler_number) candidate [meanings](https://en.wikipedia.org/wiki/Euler%E2%80%93Mascheroni_constant) for "Euler number" to make that term unambiguous), and restrict the method to an `Array{<:Integer}`, so that for a grayscale image the user first calls `label_components`. (I don't like the idea of a fallback that automatically calls `label_components` if passed an `Array{Bool}` because `Bool<:Integer`. Better to keep the concepts orthogonal.)

Sadly I don't have any papers I know off the top of my head.
